src/lib/openssl_preload/shadow_openssl_rng.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * The Shadow Simulator
+ * Copyright (c) 2010-2011, Rob Jansen
+ * See LICENSE for licensing information
+ */
+
+/* Implements an LD_PRELOAD library intended for use with Shadow. libcrypto
+ * otherwise internally uses some entropy sources that Shadow is unable to trap
+ * and emulate (such as the RDRAND instruction), making Shadow simulations of
+ * software using libcrypto non-deterministic.
+ *
+ * To use this library, set LD_PRELOAD in the target program's `environment`
+ * attribute in the Shadow simulation config file. e.g.:
```

Comment:
This would also be useful in the Tor simulation documentation (`docs/getting_started_tor.md`).

Reply:
Done.